task/jefe/src/dump.rs

```diff
@@ -75,27 +103,83 @@ pub fn claim_dump_area(base: u32) -> Result<DumpArea, DumpAgentError> {
     }
 }
 
-pub fn dump_task(base: u32, task: usize) {
-    ringbuf_entry!(Trace::Dumping(task, base));
-
+/// Setup for dumping a task (either completely or a sub-region)
+fn dump_task_setup(base: u32, full: bool) -> Result<DumpArea, DumpAgentError> {
     //
     // We need to claim a dump area.  Once it's claimed, we have committed
     // to dumping into it:  any failure will result in a partial or otherwise
     // corrupted dump.
     //
     let area = humpty::claim_dump_area(
         base,
-        DumpContents::SingleTask,
+        if full {
+            DumpContents::SingleTask
```

Comment:
I admit a mild preference for exposing this enum in place of `bool` in the function signature, as I think it'd make it clearer.

Reply:
I'll do you one better and make a _new_ type, since `DumpContents` has other variants that aren't relevant here.

Done in c61c0635bf453f533061008cd54e60e5de1fe715